src/main/java/com/uber/cadence/serviceclient/IWorkflowService.java

```diff
@@ -84,4 +86,6 @@ void SignalWorkflowExecutionWithTimeout(
       AsyncMethodCallback resultHandler,
       Long timeoutInMillis)
       throws TException;
+
+  CompletableFuture<Boolean> isHealthy();
```

Comment:
I can't think of a new name and kinda like `isHeathy`. So can you add a comment about the fact that implementation may try to resolve the unhealthy issue like reset DNS endpoint? Otherwise LGTM

Reply:
Do you have a better idea for a name?